Add example of Stock Data Anlysis (#9)
zylon-ai/zylon-docs: api-reference/examples/stock_analysis.mdx

    """
    end_date = datetime.now()
    start_date = end_date - timedelta(days=30 * months)
    df = yf.download(ticker, start=start_date.strftime("%Y-%m-%d"), end=end_date.strftime("%Y-%m-%d"), auto_adjust=True)

    if df.empty:
        raise Exception(f"No data found for ticker {ticker}")

    csv_path = f"./{ticker}_price.csv"
    df.to_csv(csv_path)
    print(f"Price data for {ticker} saved to {csv_path}")
    return csv_path

def remove_price_csv(ticker: str):
    """
    Remove the local CSV file for a given stock ticker.
    """
    csv_path = f"./{ticker}_price.csv"
    try:
        os.remove(csv_path)
        print(f"Removed local price CSV file: {csv_path}")
    except FileNotFoundError:
        print(f"No local price CSV file found to remove: {csv_path}")

def format_response(query: str, response: Dict) -> str:
    """
    Extract and format the assistant's response from the Zylon API response.
    """
    return f"""
    Query: {query}
    Response: {response["content"][0]["text"]}
    """

def extract_response_text(response: dict) -> str:
    """
    Extract the main text response from the Zylon API response.
    """
    contents = response.get("content", [])

>     for item in reversed(contents):  # Busca desde el último para capturar la respuesta final

Reviewer: ```suggestion
    for item in reversed(contents):  # Extract content from the end
```

        if isinstance(item, dict) and "text" in item:
            return item["text"]
    return ""

def format_tool_response(query: str, tool_response: Dict) -> str:
    """
    Extract and format the tool's response from the Zylon API response.
    """
    return f"""
    Query: {query}
    Tool Response: {extract_response_text(tool_response)}
    """

if __name__ == '__main__':
    ticker = "AAPL"
    example_files = [
            "https://www.apple.com/newsroom/pdfs/fy2025-q2/FY25_Q2_Consolidated_Financial_Statements.pdf",
            "https://www.apple.com/newsroom/pdfs/fy2025-q3/FY25_Q3_Consolidated_Financial_Statements.pdf"
            ]
    
    # Ingest uris documents
    documents_artifact_ids = []
    for file_url in example_files:
        artifact_id = str(uuid.uuid4())
        documents_artifact_ids.append(artifact_id)
        ingest_from_uri(artifact_id, file_url, ticker)

    ## Generate, ingest and remove price CSV
    price_csv_path = generate_price_csv(ticker)
    price_artifact_id = str(uuid.uuid4())
    ingest_from_file(price_artifact_id, price_csv_path, ticker)
    remove_price_csv(ticker)

>     ##Query documents

Reviewer: ```suggestion
    # Query documents
```
